diff -urN ganeti-3.1.0.rc2.default/lib/client/gnt_cluster.py ganeti-3.1.0.rc2/lib/client/gnt_cluster.py
--- ganeti-3.1.0.rc2.default/lib/client/gnt_cluster.py	2025-02-12 22:18:44.000000000 +0900
+++ ganeti-3.1.0.rc2/lib/client/gnt_cluster.py	2025-03-30 01:07:11.930870314 +0900
@@ -1172,12 +1172,6 @@
       ctx.ssh.CopyFileToNode(node_name, port,
                              ssconf_master_candidate_certs_filename)
 
-    # Write the boostrap entry to the config using wconfd.
-    config_live_lock = utils.livelock.LiveLock("renew_crypto")
-    cfg = config.GetConfig(None, config_live_lock)
-    cfg.AddNodeToCandidateCerts(constants.CRYPTO_BOOTSTRAP, master_digest)
-    cfg.Update(cfg.GetClusterInfo(), ctx.feedback_fn)
-
   def _RenewServerAndClientCerts(ctx):
     ctx.feedback_fn("Updating the cluster SSL certificate.")
 
