diff -urN ganeti-3.0.2.default/lib/hypervisor/hv_kvm/monitor.py ganeti-3.0.2/lib/hypervisor/hv_kvm/monitor.py
--- ganeti-3.0.2.default/lib/hypervisor/hv_kvm/monitor.py	2022-03-01 05:37:07.000000000 +0900
+++ ganeti-3.0.2/lib/hypervisor/hv_kvm/monitor.py	2022-09-24 13:43:13.353854856 +0900
@@ -128,7 +128,7 @@
 
 
 class MonitorSocket(object):
-  _SOCKET_TIMEOUT = 5
+  _SOCKET_TIMEOUT = 15
 
   def __init__(self, monitor_filename):
     """Instantiates the MonitorSocket object.
