diff -urN ganeti-2.15.0.default/daemons/daemon-util.in ganeti-2.15.0/daemons/daemon-util.in
--- ganeti-2.15.0.default/daemons/daemon-util.in	2015-07-29 16:50:04.000000000 +0900
+++ ganeti-2.15.0/daemons/daemon-util.in	2015-08-14 14:12:11.507000137 +0900
@@ -414,7 +414,11 @@
 
 # Reloads the SSH keys
 reload_ssh_keys() {
-  @RPL_SSH_INITD_SCRIPT@ restart
+  if use_systemctl; then
+    systemctl restart sshd.service
+  else
+    @RPL_SSH_INITD_SCRIPT@ restart
+  fi
 }
 
 # Read @SYSCONFDIR@/rc.d/init.d/functions if start-stop-daemon not available
