diff -urN ganeti-2.8.0.default/daemons/daemon-util.in ganeti-2.8.0/daemons/daemon-util.in
--- ganeti-2.8.0.default/daemons/daemon-util.in	2013-09-26 20:22:46.000000000 +0900
+++ ganeti-2.8.0/daemons/daemon-util.in	2013-10-08 16:06:28.406248406 +0900
@@ -354,7 +354,11 @@
 
 # Reloads the SSH keys
 reload_ssh_keys() {
-  @RPL_SSH_INITD_SCRIPT@ restart
+  if [[ -f /etc/fedora-release ]]; then
+    systemctl restart sshd.service
+  else
+    @RPL_SSH_INITD_SCRIPT@ restart
+  fi
 }
 
 # Read @SYSCONFDIR@/rc.d/init.d/functions if start-stop-daemon not available
