diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-kvmd.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-kvmd.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-kvmd.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-kvmd.service.in	2014-08-09 15:20:03.014866277 +0900
@@ -1,8 +1,6 @@
 [Unit]
 Description = Ganeti KVM daemon (kvmd)
 Documentation = man:ganeti-kvmd(8)
-Requires = ganeti-common.service
-After = ganeti-common.service
 PartOf = ganeti-noded.target
 
 [Service]
diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-luxid.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-luxid.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-luxid.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-luxid.service.in	2014-08-09 15:20:03.014866277 +0900
@@ -1,8 +1,6 @@
 [Unit]
 Description = Ganeti query daemon (luxid)
 Documentation = man:ganeti-luxid(8)
-Requires = ganeti-common.service
-After = ganeti-common.service
 PartOf = ganeti-master.target
 ConditionPathExists = @LOCALSTATEDIR@/lib/ganeti/config.data
 
diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-metad.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-metad.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-metad.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-metad.service.in	2014-08-09 15:20:03.014866277 +0900
@@ -1,7 +1,5 @@
 [Unit]
 Description = Ganeti instance metadata daemon (metad)
-Requires = ganeti-common.service
-After = ganeti-common.service
 PartOf = ganeti-noded.target
 
 [Service]
diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-mond.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-mond.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-mond.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-mond.service.in	2014-08-09 15:20:03.015866277 +0900
@@ -1,8 +1,6 @@
 [Unit]
 Description = Ganeti monitoring daemon (mond)
 Documentation = man:ganeti-mond(8)
-Requires = ganeti-common.service
-After = ganeti-common.service
 PartOf = ganeti-node.target
 
 [Service]
diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-noded.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-noded.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-noded.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-noded.service.in	2014-08-09 16:12:43.987839688 +0900
@@ -1,8 +1,8 @@
 [Unit]
 Description = Ganeti node daemon (noded)
 Documentation = man:ganeti-noded(8)
-After = ganeti-common.service
-Requires = ganeti-common.service
+Requires = ganeti-confd.service
+After = ganeti-confd.service
 PartOf = ganeti-node.target
 ConditionPathExists = @LOCALSTATEDIR@/lib/ganeti/server.pem
 
diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-rapi.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-rapi.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-rapi.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-rapi.service.in	2014-08-09 15:20:25.198867588 +0900
@@ -1,9 +1,7 @@
 [Unit]
 Description = Ganeti RAPI daemon (rapi)
 Documentation = man:ganeti-rapi(8)
-Requires = ganeti-common.service
-Requisite = ganeti-luxid.service
-After = ganeti-common.service
+Requires = ganeti-luxid.service
 PartOf = ganeti-master.target
 ConditionPathExists = @LOCALSTATEDIR@/lib/ganeti/rapi.pem
 
diff -urN ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-wconfd.service.in ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-wconfd.service.in
--- ganeti-2.12.0.beta1.default/doc/examples/systemd/ganeti-wconfd.service.in	2014-07-17 01:42:40.000000000 +0900
+++ ganeti-2.12.0.beta1/doc/examples/systemd/ganeti-wconfd.service.in	2014-08-09 15:20:03.015866277 +0900
@@ -1,8 +1,6 @@
 [Unit]
 Description = Ganeti config writer daemon (wconfd)
 Documentation = man:ganeti-wconfd(8)
-Requires = ganeti-common.service
-After = ganeti-common.service
 PartOf = ganeti-master.target
 ConditionPathExists = @LOCALSTATEDIR@/lib/ganeti/config.data
 
