diff -urN ganeti-2.14.0.default/daemons/daemon-util.in ganeti-2.14.0/daemons/daemon-util.in
--- ganeti-2.14.0.default/daemons/daemon-util.in	2015-06-01 20:30:36.000000000 +0900
+++ ganeti-2.14.0/daemons/daemon-util.in	2015-07-05 17:23:41.862026667 +0900
@@ -414,7 +414,11 @@
 
 # Reloads the SSH keys
 reload_ssh_keys() {
-  @RPL_SSH_INITD_SCRIPT@ restart
+  if use_systemctl; then
+    systemctl restart sshd.service
+  else
+    @RPL_SSH_INITD_SCRIPT@ restart
+  fi
 }
 
 # Read @SYSCONFDIR@/rc.d/init.d/functions if start-stop-daemon not available
