diff -urN ganeti-2.16.1.default/doc/examples/systemd/ganeti-metad.service.in ganeti-2.16.1/doc/examples/systemd/ganeti-metad.service.in
--- ganeti-2.16.1.default/doc/examples/systemd/ganeti-metad.service.in	2019-04-01 15:29:37.000000000 +0900
+++ ganeti-2.16.1/doc/examples/systemd/ganeti-metad.service.in	2020-02-28 13:19:40.410229662 +0900
@@ -12,6 +12,7 @@
 Restart = on-failure
 CapabilityBoundingSet=CAP_NET_BIND_SERVICE
 Capabilities=cap_net_bind_service+=ep
+AmbientCapabilities=CAP_NET_BIND_SERVICE
 
 # ganeti-metad is started on-demand by noded, so there must be no Install
 # section.
