diff -urN ganeti-3.0.0.default/cabal/ganeti.template.cabal ganeti-3.0.0/cabal/ganeti.template.cabal
--- ganeti-3.0.0.default/cabal/ganeti.template.cabal	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/cabal/ganeti.template.cabal	2021-01-19 17:52:10.955130529 +0900
@@ -69,7 +69,7 @@
     , curl                          >= 1.3.7      && < 1.4
     , hinotify                      >= 0.3.2      && < 0.5
     , hslogger                      >= 1.1.4      && < 1.4
-    , json                          >= 0.5        && < 1.0
+    , json                          >= 0.5        && < 0.10
     , lens                          >= 3.10       && < 5.0
     , lifted-base                   >= 0.2.0.3    && < 0.3
     , monad-control                 >= 0.3.1.3    && < 1.1
diff -urN ganeti-3.0.0.default/ganeti.cabal ganeti-3.0.0/ganeti.cabal
--- ganeti-3.0.0.default/ganeti.cabal	2020-12-23 21:40:15.000000000 +0900
+++ ganeti-3.0.0/ganeti.cabal	2021-01-19 17:52:00.176167837 +0900
@@ -316,7 +316,7 @@
     , curl                          >= 1.3.7      && < 1.4
     , hinotify                      >= 0.3.2      && < 0.5
     , hslogger                      >= 1.1.4      && < 1.4
-    , json                          >= 0.5        && < 1.0
+    , json                          >= 0.5        && < 0.10
     , lens                          >= 3.10       && < 5.0
     , lifted-base                   >= 0.2.0.3    && < 0.3
     , monad-control                 >= 0.3.1.3    && < 1.1
