diff -urN ganeti-2.16.0.default/daemons/daemon-util.in ganeti-2.16.0/daemons/daemon-util.in
--- ganeti-2.16.0.default/daemons/daemon-util.in	2018-09-19 22:43:16.000000000 +0900
+++ ganeti-2.16.0/daemons/daemon-util.in	2018-10-20 14:19:13.992938131 +0900
@@ -440,7 +440,11 @@
 
 # Reloads the SSH keys
 reload_ssh_keys() {
-  @RPL_SSH_INITD_SCRIPT@ restart
+  if use_systemctl; then
+    systemctl restart sshd.service
+  else
+    @RPL_SSH_INITD_SCRIPT@ restart
+  fi
 }
 
 # Read @SYSCONFDIR@/rc.d/init.d/functions if start-stop-daemon not available
