diff -urN ganeti-2.7.0.rc2.default/configure ganeti-2.7.0.rc2/configure
--- ganeti-2.7.0.rc2.default/configure	2013-05-24 20:13:12.000000000 +0900
+++ ganeti-2.7.0.rc2/configure	2013-05-27 19:27:19.015734087 +0900
@@ -4717,7 +4717,7 @@
      else
        am_py_prefix=$prefix
      fi
-     am_cv_python_pythondir=`$PYTHON -c "import sys; from distutils import sysconfig; sys.stdout.write(sysconfig.get_python_lib(0,0,prefix='$am_py_prefix'))" 2>/dev/null`
+     am_cv_python_pythondir=`$PYTHON -c "import sys; from distutils import sysconfig; sys.stdout.write(sysconfig.get_python_lib(1,0,prefix='$am_py_prefix'))" 2>/dev/null`
      case $am_cv_python_pythondir in
      $am_py_prefix*)
        am__strip_prefix=`echo "$am_py_prefix" | sed 's|.|.|g'`
