diff -urN ganeti-3.0.0.default/doc/examples/systemd/ganeti-common.service.in ganeti-3.0.0/doc/examples/systemd/ganeti-common.service.in
--- ganeti-3.0.0.default/doc/examples/systemd/ganeti-common.service.in	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/doc/examples/systemd/ganeti-common.service.in	2021-01-25 11:33:28.899975708 +0900
@@ -4,3 +4,6 @@
 [Service]
 Type = oneshot
 ExecStart = @PKGLIBDIR@/ensure-dirs
+# Disable start rate limit because ganeti-common.service is executed
+# multiple times from ganeti-*.service in ganeti.target.
+StartLimitBurst = 0
