diff -urN ganeti-2.10.3.default/Makefile.am ganeti-2.10.3/Makefile.am
--- ganeti-2.10.3.default/Makefile.am	2014-04-15 15:17:07.000000000 +0900
+++ ganeti-2.10.3/Makefile.am	2014-04-20 21:28:36.763000395 +0900
@@ -2237,7 +2237,7 @@
 	done
 	if ! test -n '$(ENABLE_MANPAGES)'; then \
 	  for man in $(manfullpath); do \
-	    $(LN_S) -f $(defaultversionedsharedir)/root$(MANDIR)/$$man $(DESTDIR)$(MANDIR)/$$man; \
+	    $(LN_S) -f $(defaultversionedsharedir)/root$(MANDIR)/$$man $(DESTDIR)$(MANDIR)/$$man.gz; \
 	  done; \
 	fi
 	for prog in $(myexeclib_scripts_basenames); do \
diff -urN ganeti-2.10.3.default/Makefile.in ganeti-2.10.3/Makefile.in
--- ganeti-2.10.3.default/Makefile.in	2014-04-15 15:17:09.000000000 +0900
+++ ganeti-2.10.3/Makefile.in	2014-04-20 21:29:23.877997814 +0900
@@ -4075,7 +4075,7 @@
 	done
 	if ! test -n '$(ENABLE_MANPAGES)'; then \
 	  for man in $(manfullpath); do \
-	    $(LN_S) -f $(defaultversionedsharedir)/root$(MANDIR)/$$man $(DESTDIR)$(MANDIR)/$$man; \
+	    $(LN_S) -f $(defaultversionedsharedir)/root$(MANDIR)/$$man $(DESTDIR)$(MANDIR)/$$man.gz; \
 	  done; \
 	fi
 	for prog in $(myexeclib_scripts_basenames); do \
