diff -urN ganeti-3.1.0.default/lib/client/gnt_cluster.py ganeti-3.1.0/lib/client/gnt_cluster.py
--- ganeti-3.1.0.default/lib/client/gnt_cluster.py	2025-08-23 06:50:27.000000000 +0900
+++ ganeti-3.1.0/lib/client/gnt_cluster.py	2025-08-26 19:04:43.322911250 +0900
@@ -2395,6 +2395,11 @@
              " has to be given")
     return 1
 
+  usertext = ("This Ganeti is installed by RPM package, so it is recommended"
+              " to upgrade using RPM. Continue?")
+  if not AskUser(usertext):
+    return 1
+
   # If we're not told to resume, verify there is no upgrade
   # in progress.
   if not opts.resume:
