diff -urN ganeti-3.0.0.default/lib/hypervisor/hv_kvm/monitor.py ganeti-3.0.0/lib/hypervisor/hv_kvm/monitor.py
--- ganeti-3.0.0.default/lib/hypervisor/hv_kvm/monitor.py	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/lib/hypervisor/hv_kvm/monitor.py	2021-01-20 18:47:05.630502887 +0900
@@ -716,7 +716,8 @@
     if self.version >= (3, 0, 0):
       arguments["max-postcopy-bandwidth"] = max_bandwidth
 
-    self.Execute("migrate-set-parameters", arguments)
+    if self.version >= (2, 4, 0):
+      self.Execute("migrate-set-parameters", arguments)
 
   @_ensure_connection
   def SetMigrationCapabilities(self, capabilities, state):
