diff -urN ganeti-3.1.0.rc2.default/lib/hypervisor/hv_kvm/__init__.py ganeti-3.1.0.rc2/lib/hypervisor/hv_kvm/__init__.py
--- ganeti-3.1.0.rc2.default/lib/hypervisor/hv_kvm/__init__.py	2025-02-12 22:18:44.000000000 +0900
+++ ganeti-3.1.0.rc2/lib/hypervisor/hv_kvm/__init__.py	2025-03-30 01:05:57.934974331 +0900
@@ -85,7 +85,7 @@
                                                 validate_vnc_parameters, \
                                                 validate_disk_parameters
 
-import ganeti.hypervisor.hv_kvm.kvm_utils as kvm_utils
+from ganeti.hypervisor.hv_kvm import kvm_utils
 
 _KVM_NETWORK_SCRIPT = pathutils.CONF_DIR + "/kvm-vif-bridge"
 _KVM_START_PAUSED_FLAG = "-S"
diff -urN ganeti-3.1.0.rc2.default/lib/hypervisor/hv_kvm/kvm_utils.py ganeti-3.1.0.rc2/lib/hypervisor/hv_kvm/kvm_utils.py
--- ganeti-3.1.0.rc2.default/lib/hypervisor/hv_kvm/kvm_utils.py	2025-02-07 21:33:21.000000000 +0900
+++ ganeti-3.1.0.rc2/lib/hypervisor/hv_kvm/kvm_utils.py	2025-03-30 01:05:54.524748704 +0900
@@ -59,7 +59,8 @@
   @param uri: storage-describing URI
   @return: dict
   """
-  if (match := re.match(_BLOCKDEV_URI_REGEX_GLUSTER, uri)) is not None:
+  match = re.match(_BLOCKDEV_URI_REGEX_GLUSTER, uri)
+  if match is not None:
     return {
         "driver": "gluster",
         "server": [
@@ -72,7 +73,8 @@
         "volume": match.group("volume"),
         "path": match.group("path")
       }
-  elif (match := re.match(_BLOCKDEV_URI_REGEX_RBD, uri)) is not None:
+  match = re.match(_BLOCKDEV_URI_REGEX_RBD, uri)
+  if match is not None:
     return {
         "driver": "rbd",
         "pool": match.group("pool"),
diff -urN ganeti-3.1.0.rc2.default/lib/hypervisor/hv_kvm/monitor.py ganeti-3.1.0.rc2/lib/hypervisor/hv_kvm/monitor.py
--- ganeti-3.1.0.rc2.default/lib/hypervisor/hv_kvm/monitor.py	2025-02-12 22:18:44.000000000 +0900
+++ ganeti-3.1.0.rc2/lib/hypervisor/hv_kvm/monitor.py	2025-03-30 01:05:57.934974331 +0900
@@ -48,7 +48,7 @@
 from ganeti import constants
 from ganeti import serializer
 
-import ganeti.hypervisor.hv_kvm.kvm_utils as kvm_utils
+from ganeti.hypervisor.hv_kvm import kvm_utils
 
 
 class QmpCommandNotSupported(errors.HypervisorError):
