diff -urN ganeti-2.16.1.default/cabal/ganeti.template.cabal ganeti-2.16.1/cabal/ganeti.template.cabal
--- ganeti-2.16.1.default/cabal/ganeti.template.cabal	2019-04-01 15:29:37.000000000 +0900
+++ ganeti-2.16.1/cabal/ganeti.template.cabal	2019-04-16 17:41:15.156262034 +0900
@@ -53,7 +53,7 @@
     , pretty                        >= 1.1.1.0
     , process                       >= 1.1.0.1
     , random                        >= 1.0.1.1
-    , semigroups                    >= 0.15.0
+    , semigroups                    >= 0.8.5
     , template-haskell              >= 2.7.0.0
     , text                          >= 0.11.1.13
     , transformers                  >= 0.3.0.0
