diff -urN ganeti-3.1.0.rc2.default/lib/client/gnt_cluster.py ganeti-3.1.0.rc2/lib/client/gnt_cluster.py
--- ganeti-3.1.0.rc2.default/lib/client/gnt_cluster.py	2025-02-12 22:18:44.000000000 +0900
+++ ganeti-3.1.0.rc2/lib/client/gnt_cluster.py	2025-03-30 01:08:14.304997657 +0900
@@ -2395,6 +2395,11 @@
              " has to be given")
     return 1
 
+  usertext = ("This Ganeti is installed by RPM package, so it is recommended"
+              " to upgrade using RPM. Continue?")
+  if not AskUser(usertext):
+    return 1
+
   # If we're not told to resume, verify there is no upgrade
   # in progress.
   if not opts.resume:
