diff -urN ganeti-3.0.0.default/lib/runtime.py ganeti-3.0.0/lib/runtime.py
--- ganeti-3.0.0.default/lib/runtime.py	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/lib/runtime.py	2021-01-25 13:41:04.568419332 +0900
@@ -95,6 +95,11 @@
   @ivar rapi_uid: The resolved uid of the rapi user
   @ivar rapi_gid: The resolved gid of the rapi group
   @ivar noded_uid: The resolved uid of the noded user
+  @ivar noded_gid: The resolved uid of the noded group
+  @ivar mond_uid: The resolved uid of the mond user
+  @ivar mond_gid: The resolved gid of the mond group
+  @ivar metad_uid: The resolved uid of the metad user
+  @ivar metad_gid: The resolved gid of the metad group
   @ivar daemons_gid: The resolved gid of the daemons group
   @ivar admin_gid: The resolved gid of the admin group
 
@@ -125,6 +130,9 @@
     self.mond_uid = GetUid(constants.MOND_USER, _getpwnam)
     self.mond_gid = GetGid(constants.MOND_GROUP, _getgrnam)
 
+    self.metad_uid = GetUid(constants.METAD_USER, _getpwnam)
+    self.metad_gid = GetGid(constants.METAD_GROUP, _getgrnam)
+
     # Misc Ganeti groups
     self.daemons_gid = GetGid(constants.DAEMONS_GROUP, _getgrnam)
     self.admin_gid = GetGid(constants.ADMIN_GROUP, _getgrnam)
@@ -136,6 +144,7 @@
       self.luxid_uid: constants.LUXID_USER,
       self.rapi_uid: constants.RAPI_USER,
       self.noded_uid: constants.NODED_USER,
+      self.metad_uid: constants.METAD_USER,
       self.mond_uid: constants.MOND_USER,
       }
 
@@ -147,6 +156,7 @@
       self.rapi_gid: constants.RAPI_GROUP,
       self.noded_gid: constants.NODED_GROUP,
       self.mond_gid: constants.MOND_GROUP,
+      self.metad_gid: constants.METAD_GROUP,
       self.daemons_gid: constants.DAEMONS_GROUP,
       self.admin_gid: constants.ADMIN_GROUP,
       }
diff -urN ganeti-3.0.0.default/lib/tools/ensure_dirs.py ganeti-3.0.0/lib/tools/ensure_dirs.py
--- ganeti-3.0.0.default/lib/tools/ensure_dirs.py	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/lib/tools/ensure_dirs.py	2021-01-25 15:46:12.042540078 +0900
@@ -131,9 +131,16 @@
   masterd_log = constants.DAEMONS_LOGFILES[constants.MASTERD]
   noded_log = constants.DAEMONS_LOGFILES[constants.NODED]
   confd_log = constants.DAEMONS_LOGFILES[constants.CONFD]
+  wconfd_log = constants.DAEMONS_LOGFILES[constants.WCONFD]
   luxid_log = constants.DAEMONS_LOGFILES[constants.LUXID]
   rapi_log = constants.DAEMONS_LOGFILES[constants.RAPI]
   mond_log = constants.DAEMONS_LOGFILES[constants.MOND]
+  metad_log = constants.DAEMONS_LOGFILES[constants.METAD]
+
+  mond_extra_log = constants.DAEMONS_EXTRA_LOGFILES[constants.MOND]
+  metad_extra_log = constants.DAEMONS_EXTRA_LOGFILES[constants.METAD]
+
+  jobs_log = pathutils.GetLogFilename("jobs")
 
   rapi_dir = os.path.join(pathutils.DATA_DIR, "rapi")
   cleaner_log_dir = os.path.join(pathutils.LOG_DIR, "cleaner")
@@ -207,10 +214,17 @@
     (pathutils.LOG_DIR, DIR, 0o770, getent.masterd_uid, getent.daemons_gid),
     (masterd_log, FILE, 0o600, getent.masterd_uid, getent.masterd_gid, False),
     (confd_log, FILE, 0o600, getent.confd_uid, getent.masterd_gid, False),
+    (wconfd_log, FILE, 0o600, getent.wconfd_uid, getent.masterd_gid, False),
     (luxid_log, FILE, 0o600, getent.luxid_uid, getent.masterd_gid, False),
     (noded_log, FILE, 0o600, getent.noded_uid, getent.masterd_gid, False),
     (rapi_log, FILE, 0o600, getent.rapi_uid, getent.masterd_gid, False),
     (mond_log, FILE, 0o600, getent.mond_uid, getent.masterd_gid, False),
+    (mond_extra_log["access"], FILE, 0o600, getent.mond_uid, getent.masterd_gid, False),
+    (mond_extra_log["error"], FILE, 0o600, getent.mond_uid, getent.masterd_gid, False),
+    (metad_log, FILE, 0o600, getent.metad_uid, getent.metad_gid, False),
+    (metad_extra_log["access"], FILE, 0o600, getent.metad_uid, getent.metad_gid, False),
+    (metad_extra_log["error"], FILE, 0o600, getent.metad_uid, getent.metad_gid, False),
+    (jobs_log, FILE, 0o600, getent.luxid_uid, getent.luxid_gid, False),
     (pathutils.LOG_OS_DIR, DIR, 0o750, getent.noded_uid, getent.daemons_gid),
     (pathutils.LOG_XEN_DIR, DIR, 0o750, getent.noded_uid, getent.daemons_gid),
     (pathutils.LOG_KVM_DIR, DIR, 0o750, getent.noded_uid, getent.daemons_gid),
diff -urN ganeti-3.0.0.default/src/Ganeti/Constants.hs ganeti-3.0.0/src/Ganeti/Constants.hs
--- ganeti-3.0.0.default/src/Ganeti/Constants.hs	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/src/Ganeti/Constants.hs	2021-01-25 13:53:04.901817647 +0900
@@ -438,6 +438,10 @@
                  [ ("access", Runtime.daemonsExtraLogbase GanetiMond AccessLog)
                  , ("error", Runtime.daemonsExtraLogbase GanetiMond ErrorLog)
                  ])
+  , (GanetiMetad, Map.fromList
+                 [ ("access", Runtime.daemonsExtraLogbase GanetiMetad AccessLog)
+                 , ("error", Runtime.daemonsExtraLogbase GanetiMetad ErrorLog)
+                 ])
   ]
 
 extraLogreasonAccess :: String
diff -urN ganeti-3.0.0.default/test/py/ganeti.runtime_unittest.py ganeti-3.0.0/test/py/ganeti.runtime_unittest.py
--- ganeti-3.0.0.default/test/py/ganeti.runtime_unittest.py	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/test/py/ganeti.runtime_unittest.py	2021-01-25 13:39:13.642825266 +0900
@@ -52,6 +52,8 @@
     constants.NODED_USER: _EntStub(uid=3),
     constants.LUXID_USER: _EntStub(uid=4),
     constants.WCONFD_USER: _EntStub(uid=5),
+    constants.MOND_USER: _EntStub(uid=6),
+    constants.METAD_USER: _EntStub(uid=7),
     }
   return users[user]
 
@@ -66,6 +68,8 @@
     constants.NODED_GROUP: _EntStub(gid=5),
     constants.LUXID_GROUP: _EntStub(gid=6),
     constants.WCONFD_GROUP: _EntStub(gid=7),
+    constants.MOND_GROUP: _EntStub(gid=8),
+    constants.METAD_GROUP: _EntStub(gid=9),
     }
   return groups[group]
 
@@ -101,12 +105,26 @@
                      _StubGetpwnam(constants.WCONFD_USER).pw_uid)
     self.assertEqual(self.resolver.wconfd_gid,
                      _StubGetgrnam(constants.WCONFD_GROUP).gr_gid)
+    self.assertEqual(self.resolver.luxid_uid,
+                     _StubGetpwnam(constants.LUXID_USER).pw_uid)
+    self.assertEqual(self.resolver.luxid_gid,
+                     _StubGetgrnam(constants.LUXID_GROUP).gr_gid)
     self.assertEqual(self.resolver.rapi_uid,
                      _StubGetpwnam(constants.RAPI_USER).pw_uid)
     self.assertEqual(self.resolver.rapi_gid,
                      _StubGetgrnam(constants.RAPI_GROUP).gr_gid)
     self.assertEqual(self.resolver.noded_uid,
                      _StubGetpwnam(constants.NODED_USER).pw_uid)
+    self.assertEqual(self.resolver.noded_gid,
+                     _StubGetgrnam(constants.NODED_GROUP).gr_gid)
+    self.assertEqual(self.resolver.mond_uid,
+                     _StubGetpwnam(constants.MOND_USER).pw_uid)
+    self.assertEqual(self.resolver.mond_gid,
+                     _StubGetgrnam(constants.MOND_GROUP).gr_gid)
+    self.assertEqual(self.resolver.metad_uid,
+                     _StubGetpwnam(constants.METAD_USER).pw_uid)
+    self.assertEqual(self.resolver.metad_gid,
+                     _StubGetgrnam(constants.METAD_GROUP).gr_gid)
 
     self.assertEqual(self.resolver.daemons_gid,
                      _StubGetgrnam(constants.DAEMONS_GROUP).gr_gid)
