diff -urN ganeti-3.0.0.default/lib/pathutils.py ganeti-3.0.0/lib/pathutils.py
--- ganeti-3.0.0.default/lib/pathutils.py	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/lib/pathutils.py	2021-01-25 15:52:46.366155240 +0900
@@ -106,6 +106,8 @@
 SSCONF_LOCK_FILE = LOCK_DIR + "/ganeti-ssconf.lock"
 
 CLUSTER_CONF_FILE = DATA_DIR + "/config.data"
+LOCK_STATUS_FILE = DATA_DIR + "/locks.data"
+TEMP_RES_STATUS_FILE = DATA_DIR + "/tempres.data"
 RAPI_CERT_FILE = DATA_DIR + "/rapi.pem"
 CONFD_HMAC_KEY = DATA_DIR + "/hmac.key"
 SPICE_CERT_FILE = DATA_DIR + "/spice.pem"
diff -urN ganeti-3.0.0.default/lib/tools/ensure_dirs.py ganeti-3.0.0/lib/tools/ensure_dirs.py
--- ganeti-3.0.0.default/lib/tools/ensure_dirs.py	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/lib/tools/ensure_dirs.py	2021-01-25 15:53:23.201025755 +0900
@@ -148,6 +148,10 @@
      getent.masterd_uid, getent.masterd_gid, False),
     (pathutils.CLUSTER_CONF_FILE, FILE, 0o640,
      getent.masterd_uid, getent.confd_gid, False),
+    (pathutils.LOCK_STATUS_FILE, FILE, 0o640,
+     getent.masterd_uid, getent.confd_gid, False),
+    (pathutils.TEMP_RES_STATUS_FILE, FILE, 0o640,
+     getent.masterd_uid, getent.confd_gid, False),
     (pathutils.CONFD_HMAC_KEY, FILE, 0o440,
      getent.confd_uid, getent.masterd_gid, False),
     (pathutils.SSH_KNOWN_HOSTS_FILE, FILE, 0o644,
