diff -urN ganeti-3.0.0.default/doc/examples/systemd/ganeti-metad.service.in ganeti-3.0.0/doc/examples/systemd/ganeti-metad.service.in
--- ganeti-3.0.0.default/doc/examples/systemd/ganeti-metad.service.in	2020-12-23 21:40:12.000000000 +0900
+++ ganeti-3.0.0/doc/examples/systemd/ganeti-metad.service.in	2021-01-25 11:37:31.018137781 +0900
@@ -13,7 +13,7 @@
 ExecStart = @SBINDIR@/ganeti-metad -f $METAD_ARGS $ONETIME_ARGS
 Restart = on-failure
 CapabilityBoundingSet=CAP_NET_BIND_SERVICE
-Capabilities=cap_net_bind_service+=ep
+AmbientCapabilities=CAP_NET_BIND_SERVICE
 
 # ganeti-metad is started on-demand by noded, so there must be no Install
 # section.
