diff -urN snf-image-0.23.1.default/snf-image-host/snf-image.conf.in snf-image-0.23.1/snf-image-host/snf-image.conf.in
--- snf-image-0.23.1.default/snf-image-host/snf-image.conf.in	2018-01-23 21:32:44.000000000 +0900
+++ snf-image-0.23.1/snf-image-host/snf-image.conf.in	2019-03-18 13:58:14.499985064 +0900
@@ -119,7 +119,7 @@
 # Paths for needed programs. Uncomment and change the variables below if you
 # don't want to use the default one.
 # MD5SUM="md5sum"
-# KVM="kvm"
+KVM="/usr/libexec/qemu-kvm"
 # LOSETUP="losetup"
 # TIMEOUT="timeout"
 # CURL="curl"
