diff -urN dump-0.4b47.default/configure.ac dump-0.4b47/configure.ac
--- dump-0.4b47.default/configure.ac	2021-01-02 01:44:21.000000000 +0900
+++ dump-0.4b47/configure.ac	2025-08-27 13:02:43.580177124 +0900
@@ -487,7 +487,6 @@
 	common/Makefile
 	compat/lib/Makefile
 	dump/Makefile
-	restore/Makefile
 	rmt/Makefile
 ]))
 AC_OUTPUT
diff -urN dump-0.4b47.default/Makefile.am dump-0.4b47/Makefile.am
--- dump-0.4b47.default/Makefile.am	2021-01-02 00:05:15.000000000 +0900
+++ dump-0.4b47/Makefile.am	2025-08-27 13:02:56.728013928 +0900
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = compat/lib common dump restore
+SUBDIRS = compat/lib common dump
 if BUILD_RMT
 SUBDIRS += rmt
 endif
